mbotmake2/extract_thumbnails.py

```diff
@@ -8,23 +8,68 @@
 
 def extractThumbnails(filename: Path, output_dir: Path) -> list[Path]:
     with open(filename, "r") as file:
+        # Read the file in chunks to extract only relevant parts containing THUMBNAIL_BLOCK_START or THUMBNAIL_BLOCK_END
+        # Load first 5MB of the file
+        file.seek(0)
+        content = file.read(5_000_000)
+
+        # It seems that these are pretty standard for thumbnail blocks
+        # Orcaslicer uses '; THUMBNAIL_BLOCK_START' and '; THUMBNAIL_BLOCK_END' additionally to wrap the thumbnail block.
+        THUMBNAIL_BLOCK_START = "; thumbnail begin"
+        THUMBNAIL_BLOCK_END   = "; thumbnail end"
+
+        currentlyCollectingThumbnail = False
+        countCollectedThumbnail = 0
+        relevant_parts = []
+        for line in content.splitlines():
+            if THUMBNAIL_BLOCK_START in line:
+                currentlyCollectingThumbnail = True
+                #print(f"Found '{THUMBNAIL_BLOCK_START}'. Starting to collect thumbnail data.")
+
+            if currentlyCollectingThumbnail:
+                if line.strip() and line.strip() != ";":
+                    relevant_parts.append(line)
+
+            if THUMBNAIL_BLOCK_END in line:
+                currentlyCollectingThumbnail = False
+                countCollectedThumbnail += 1
```

Comment:
Ah I see what we are doing here... so this for-loop tries to defeat the parsing mechanism at `mbotmake2/grammars/thumbnails.py`. Let me show you how it is done in the other pull request.